clarity_ext/domain/container.py

```diff
@@ -180,8 +187,16 @@ def set_well(self, well_pos, artifact_name=None, artifact_id=None, artifact=None
         # TODO: Accept only an artifact
         self.wells[well_pos].artifact = artifact
 
+    def __iter__(self):
```

Comment:
Yes, so I believe these are not exactly the same use cases, since in your case you're yielding a generator in `outer_yield`, while in my case I'm looping over a generator and yielding its content. I think I've seen somewhere that your example should work as expected in Python 3.

However, I don't need to loop over the generator, as you point out. I'll need to do some fixes so I'll change this too right away.